backend/src/transactions/user.go

```diff
@@ -16,3 +16,43 @@ func GetAllUsers(db *gorm.DB) ([]models.User, error) {
 
 	return users, nil
 }
+
+func GetUser(db *gorm.DB, id uint) (*models.User, error) {
+	var user models.User
+	if err := db.First(&user, id).Error; err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, fiber.NewError(fiber.StatusNotFound, err.Error())
+		}
+
+		return nil, fiber.NewError(fiber.StatusInternalServerError, err.Error())
+	}
+
+	return &user, nil
+}
+
+func CreateUser(db *gorm.DB, user *models.User) (*models.User, error) {
```

Comment:
do we need these separate queries or can we allow the DB unique constraints to handle this for us?

Reply:
Just didn't want to throw 500 errors for things related to uniqueness, imo that's a 400